behaviour/graql/language/undefine.feature

```diff
@@ -260,45 +1218,286 @@ Feature: Graql Undefine Query
       | x      |
       | RULE   |
       | A_RULE |
-    Then graql undefine
+    When graql undefine
       """
       undefine a-rule sub rule;
       """
+    Then the integrity is validated
     Then get answers of graql query
       """
-      match $x sub rule; get;  
+      match $x sub rule; get;
       """
     Then answer size is: 1
 
 
-  Scenario: undefine a supertype errors if subtypes exist
+  Scenario: after undefining a rule, concepts previously inferred by that rule are no longer inferred
     Given graql define
       """
-      define child sub person;  
+      define
+      samuel-email-rule sub rule, when {
+        $x has email "samuel@grakn.ai";
+      }, then {
+        $x has name "Samuel";
+      };
       """
-    Then graql undefine throws
+    Given the integrity is validated
+    Given graql insert
       """
-      undefine person sub entity;
+      insert $x isa person, has email "samuel@grakn.ai";
+      """
+    Given the integrity is validated
+    Given get answers of graql query
+      """
+      match
+        $x has name $n;
+      get $n;
+      """
+    Given concept identifiers are
+      |     | check | value       |
+      | SAM | value | name:Samuel |
+    Given uniquely identify answer concepts
+      | n   |
+      | SAM |
+    When graql undefine
+      """
+      undefine samuel-email-rule sub rule;
       """
     Then the integrity is validated
+    When get answers of graql query
+      """
+      match
+        $x has name $n;
+      get $n;
+      """
+    Then answer size is: 0
+
 
+  ############
+  # ABSTRACT #
```

Comment:
double check these now with the new thoughts :)

Reply:
Implemented:

`Scenario: an abstract type can be changed into a concrete type even if has an abstract child type`
`Scenario: undefining abstract on an attribute type is allowed, even if that attribute type has an owner`